rubocop-klaxit/cops/active-record/no_transaction_on_add_index_migration.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module ActiveRecord
+      # Ensure migrations which add indexes are always outside transactions
+      # with disable_ddl_transaction!
+      class NoTransactionOnAddIndexMigration < Cop
+        MSG = "Migrations which add indexes should not be in transactions"
+
+        def_node_matcher :is_migration?, <<~PATTERN
+          (class
+            <
+              (send
+                (const (const nil? :ActiveRecord) :Migration)
+                ...)
+              ...
+            >)
+        PATTERN
+
+        def_node_search :has_send_with_sym?, <<~PATTERN
+          (send nil? %1 ...)
+        PATTERN
+
+        def on_class(node)
+          return unless is_migration?(node)
+          return if has_send_with_sym?(node, :add_index) &&
+                    has_send_with_sym?(node, :disable_ddl_transaction!)
+
+          add_offense(node)
```

Comment:
If none are true you would add an offence for every migration, wouldn't you ?

Reply:
my bad, corrected!